controlplane/kubeadm/controllers/suite_test.go

```diff
@@ -20,14 +20,34 @@ import (
 	"os"
 	"testing"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/cluster-api/internal/envtest"
 	ctrl "sigs.k8s.io/controller-runtime"
 	// +kubebuilder:scaffold:imports
 )
 
 var (
-	env *envtest.Environment
-	ctx = ctrl.SetupSignalHandler()
+	env                           *envtest.Environment
+	ctx                           = ctrl.SetupSignalHandler()
+	fakeGenericMachineTemplateCRD = &apiextensionsv1.CustomResourceDefinition{
```

Comment:
@fabriziopandini @killianmuldoon 

This CRD is now required to be able to run the convert api contract code in the KCP unit tests. I'm not sure if there's a better place to put it. I didn't want to refactor the tests to use testtypes in this PR.

Reply:
let's open an issue to track this code cleanup

Reply:
I could also add a TODO the same way as in: https://github.com/kubernetes-sigs/cluster-api/blob/362f129583b16bb71cabea75f35b684eaa20d7dc/controllers/topology/object_builders_test.go#L28

Probably easier to grep for those TODOs.

Reply:
I added the TODO for now. @killianmuldoon should I create an issue to track the `// TODO: move under internal/testtypes` TODOs (i.e. that they are done after the test package is ready). Or do you want to track it as part of your ongoing test refactoring?

Reply:
I think the TODO: should be enough for now - I don't think we'll be refactoring these tests until the object builders are quite stable, and it's likely to be across the code here.

Reply:
Once the code object builders is merged I'll open issues for different packages to convert the tests. Once we have one or two done they'll hopefully be relatively easy to pick up and do the changes.